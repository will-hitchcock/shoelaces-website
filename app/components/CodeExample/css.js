import colors from 'utils/colors'

const editor = {
  overflow: 'scroll'
}

const error = {
  borderLeft: `5px ${colors.darkRed} solid`,
  backgroundColor: colors.red,
  padding: '0.5em',
  color: colors.darkRed,
  marginTop: '0.5em'
}

export { editor, error }
