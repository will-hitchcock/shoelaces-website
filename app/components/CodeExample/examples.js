import ReactShoelaces from 'react-shoelaces'
// import { container } from 'shoelaces'
import Box from 'components/Box'

const ReactScope = { Box, ...ReactShoelaces }

const examples = [
  {
    name: 'basic',
    scope: ReactScope,
    code: `<Container>
  <Row>
    <Col xs={1 / 5}><Box /></Col>
    <Col xs={1 / 5}><Box /></Col>
    <Col xs={1 / 5}><Box /></Col>
    <Col xs={1 / 5}><Box /></Col>
    <Col xs={1 / 5}><Box /></Col>
  </Row>
  <Row>
    <Col xs={1 / 3}><Box /></Col>
    <Col xs={1 / 3}><Box /></Col>
    <Col xs={1 / 3}><Box /></Col>
  </Row>
  <Row>
    <Col xs={1 / 2}><Box /></Col>
    <Col xs={1 / 2}><Box /></Col>
  </Row>
</Container>
  `
  },
  {
    name: 'justify',
    scope: ReactScope,
    code: `<Container>
  <Row>
    <Col xs={1 / 3}><Box /></Col>
    <Col xs={1 / 3}><Box /></Col>
  </Row>
  <Row justifyContent="around">
    <Col xs={1 / 3}><Box /></Col>
    <Col xs={1 / 3}><Box /></Col>
  </Row>
  <Row justifyContent="center">
    <Col xs={1 / 3}><Box /></Col>
    <Col xs={1 / 3}><Box /></Col>
  </Row>
  <Row justifyContent="start">
    <Col xs={1 / 3}><Box /></Col>
    <Col xs={1 / 3}><Box /></Col>
  </Row>
  <Row justifyContent="end">
    <Col xs={1 / 3}><Box /></Col>
    <Col xs={1 / 3}><Box /></Col>
  </Row>
</Container>
  `
  },
  {
    name: 'nesting',
    scope: ReactScope,
    code: `<Container>
  <Row>
    <Col xs={1 / 2}>
      <Box color="primaryDark">
        <Row>
          <Col xs={1 / 3}><Box/></Col>
          <Col xs={2 / 3}>
            <Box>
              <Row>
                <Col xs={1 / 4}><Box color="primaryDarker"/></Col>
                <Col xs={3 / 4}><Box color="primaryDarker"/></Col>
              </Row>
            </Box>
          </Col>
        </Row>
      </Box>
    </Col>
    <Col xs={1 / 2}><Box color="primaryDark" /></Col>
  </Row>
</Container>
  `
  },
  {
    name: 'responsive',
    scope: ReactScope,
    code: `<Container>
  <Row>
    <Col xs={1} md={1 / 2} lg={1 / 4}><Box /></Col>
    <Col xs={1} md={1 / 2} lg={1 / 4}><Box /></Col>
    <Col xs={1} md={1 / 2} lg={1 / 4}><Box /></Col>
    <Col xs={1} md={1 / 2} lg={1 / 4}><Box /></Col>
  </Row>
  <Row>
    <Col xs={1} lg={1 / 3}><Box /></Col>
    <Col xs={1} lg={1 / 3}><Box /></Col>
    <Col xs={1} lg={1 / 3}><Box /></Col>
  </Row>
</Container>
  `
  },
  {
    name: 'math',
    scope: ReactScope,
    code: `<Container>
  <Row>
    <Col xs={1} md={1 / Math.PI}><Box /></Col>
    <Col xs={1} md={1 / 1 - (1 / Math.PI)}><Box /></Col>
  </Row>
  <Row>
    <Col xs={1} md={1 / 1 - (1 / Math.E)}><Box /></Col>
    <Col xs={1} md={1 / Math.E}><Box /></Col>
  </Row>
  <Row>
    <Col xs={1} md={1 - (1 / ((1 + Math.sqrt(5)) / 2))}><Box /></Col>
    <Col xs={1} md={1 / ((1 + Math.sqrt(5)) / 2)}><Box /></Col>
  </Row>
</Container>
`
  }
]

export default examples
