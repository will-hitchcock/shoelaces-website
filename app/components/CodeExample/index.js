/**
*
* CodeExample
*
*/
import React from 'react'
import styled from 'styled-components'
import { string, object } from 'prop-types'
import { LiveProvider, LiveEditor, LiveError, LivePreview } from 'react-live'
import { Row, Col } from 'react-shoelaces'
import { flex } from 'shoelaces'
import { editor, error } from './css'

const StyledEditor = styled(LiveEditor)`
  ${editor}
  ${flex}`
const StyledError = styled(LiveError)`
  ${error}
`

const CodeExample = ({ code, scope }) =>
  <LiveProvider scope={scope} code={code}>
    <LivePreview />
    <Row justifyContent="center">
      <Col xs={1} lg={8 / 12}>
        <StyledEditor />
        <StyledError />
      </Col>
    </Row>
  </LiveProvider>


CodeExample.propTypes = {
  code: string,
  scope: object
}

export default CodeExample
