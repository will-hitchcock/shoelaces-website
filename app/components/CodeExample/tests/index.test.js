import React from 'react'
import CodeExample from '../index'

describe('<CodeExample />', () => {
  it('renders a <LiveProvider />', () => {
    expect(shallow(
      <CodeExample scope={{ pi: Math.PI, e: Math.E }} code={'<p>pi</p>'} />
    )).toMatchSnapshot()
  })
})
