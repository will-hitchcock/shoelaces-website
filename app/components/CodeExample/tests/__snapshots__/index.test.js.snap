exports[`<CodeExample /> renders a <LiveProvider /> 1`] = `
<LiveProvider
  code="<p>pi</p>"
  mountStylesheet={true}
  noInline={false}
  scope={
    Object {
      "e": 2.718281828459045,
      "pi": 3.141592653589793,
    }
  }>
  <LivePreview />
  <styled.div
    justifyContent="center">
    <styled.div
      lg={0.6666666666666666}
      xs={1}>
      <Styled(LiveEditor) />
      <Styled(LiveError) />
    </styled.div>
  </styled.div>
</LiveProvider>
`;
